File: src/main/java/com/drunkensoftball/api/auth/service/AuthenticationService.java
package com.drunkensoftball.api.auth.service;

import com.drunkensoftball.api.auth.domain.Authentication;
import com.drunkensoftball.api.user.domain.User;

public interface AuthenticationService {

    User getUserFromToken(String token);

    Authentication createToken(User user);

    Authentication googleAuthentication(String googleToken);

    Authentication basicAuthentication(String user, String password);


}

File: src/main/kotlin/com/drunkensoftball/api/auth/service/AuthenticationService.kt
package com.drunkensoftball.api.auth.service

import com.drunkensoftball.api.auth.domain.AuthenticationEntity
import com.drunkensoftball.api.user.domain.User

interface AuthenticationService {

    fun getUserFromToken(token: String): User

    fun googleAuthentication(googleToken: String): AuthenticationEntity

    fun loginWithBasicAuthentication(user: String,
                                     password: String): AuthenticationEntity

    fun createNewUserBasicAuthentication(username: String,
                                         password: String,
                                         email: String,
                                         firstName: String?,
                                         lastName: String?): AuthenticationEntity


}
